File: apps/mhc/mobile-app/lib/models/medecin_conseil.dart
/// Coordonnées d'un médecin-conseil (snapshot affichable hors ligne).
class MedecinConseilContact {
  const MedecinConseilContact({
    required this.id,
    this.nom,
    this.telephone,
    this.email,
  });

  final int id;
  final String? nom;
  final String? telephone;
  final String? email;

  bool get hasAnyContact =>
      (nom != null && nom!.trim().isNotEmpty) ||
      (telephone != null && telephone!.trim().isNotEmpty) ||
      (email != null && email!.trim().isNotEmpty);

  factory MedecinConseilContact.fromJson(Map<String, dynamic> json) {
    return MedecinConseilContact(
      id: json['id'] is int ? json['id'] as int : int.tryParse('${json['id']}') ?? 0,
      nom: (json['nom'] as String?)?.trim(),
      telephone: (json['telephone'] as String?)?.trim(),
      email: (json['email'] as String?)?.trim(),
    );
  }

  Map<String, dynamic> toJson() => {
        'id': id,
        'nom': nom,
        'telephone': telephone,
        'email': email,
      };
}

/// Association souscription / destination / médecin-conseil.
class MedecinConseilAssignment {
  const MedecinConseilAssignment({
    required this.souscriptionId,
    this.numeroSouscription,
    this.statutSouscription,
    this.destination,
    this.destinationCountryId,
    this.destinationCountryName,
    this.medecinConseil,
  });

  final int souscriptionId;
  final String? numeroSouscription;
  final String? statutSouscription;
  final String? destination;
  final int? destinationCountryId;
  final String? destinationCountryName;
  final MedecinConseilContact? medecinConseil;

  bool get hasContact => medecinConseil?.hasAnyContact == true;

  String get destinationLabel {
    final display = destination?.trim();
    if (display != null && display.isNotEmpty) return display;
    return destinationCountryName?.trim().isNotEmpty == true
        ? destinationCountryName!.trim()
        : 'Destination non renseignée';
  }

  factory MedecinConseilAssignment.fromJson(Map<String, dynamic> json) {
    final raw = json['medecin_conseil'];
    return MedecinConseilAssignment(
      souscriptionId: json['souscription_id'] is int
          ? json['souscription_id'] as int
          : int.tryParse('${json['souscription_id']}') ?? 0,
      numeroSouscription: json['numero_souscription'] as String?,
      statutSouscription: json['statut_souscription'] as String?,
      destination: json['destination'] as String?,
      destinationCountryId: json['destination_country_id'] is int
          ? json['destination_country_id'] as int
          : int.tryParse('${json['destination_country_id'] ?? ''}'),
      destinationCountryName: json['destination_country_name'] as String?,
      medecinConseil: raw is Map<String, dynamic> ? MedecinConseilContact.fromJson(raw) : null,
    );
  }

  Map<String, dynamic> toJson() => {
        'souscription_id': souscriptionId,
        'numero_souscription': numeroSouscription,
        'statut_souscription': statutSouscription,
        'destination': destination,
        'destination_country_id': destinationCountryId,
        'destination_country_name': destinationCountryName,
        'medecin_conseil': medecinConseil?.toJson(),
      };
}

File: apps/mhc/frontend-simple/js/medecin-conseil.js
const MEDECIN_CONSEIL_CACHE_KEY = 'mhc_medecin_conseil_assignments_v1';

function escapeMedecinConseilHtml(value) {
    if (value === null || value === undefined) return '';
    const div = document.createElement('div');
    div.textContent = String(value);
    return div.innerHTML;
}

function readMedecinConseilCache() {
    try {
        const raw = localStorage.getItem(MEDECIN_CONSEIL_CACHE_KEY);
        if (!raw) return [];
        const parsed = JSON.parse(raw);
        return Array.isArray(parsed) ? parsed : [];
    } catch (_) {
        return [];
    }
}

function writeMedecinConseilCache(items) {
    try {
        localStorage.setItem(MEDECIN_CONSEIL_CACHE_KEY, JSON.stringify(items || []));
    } catch (_) {
        // Quota ou navigation privée : l'affichage en ligne reste disponible.
    }
}

async function fetchMedecinConseilAssignments(souscriptionId) {
    const query = souscriptionId ? `?souscription_id=${encodeURIComponent(souscriptionId)}` : '';
    const items = await apiCall(`/sos/medecin-conseil${query}`);
    const list = Array.isArray(items) ? items : [];
    if (!souscriptionId) {
        writeMedecinConseilCache(list);
    }
    return list;
}

async function loadMedecinConseilAssignments({ souscriptionId = null, preferCache = true } = {}) {
    const cached = readMedecinConseilCache();
    const cachedForContext = souscriptionId
        ? cached.filter((item) => String(item.souscription_id) === String(souscriptionId))
        : cached;
    try {
        const fresh = await fetchMedecinConseilAssignments(souscriptionId);
        return { items: fresh, fromCache: false };
    } catch (_) {
        return { items: preferCache ? cachedForContext : [], fromCache: cachedForContext.length > 0 };
    }
}

function renderMedecinConseilCard(assignment) {
    const contact = assignment?.medecin_conseil || null;
    const destination = assignment?.destination || assignment?.destination_country_name || 'Destination non renseignée';
    const nom = contact?.nom || 'Médecin-conseil non renseigné';
    const telephone = contact?.telephone || '';
    const email = contact?.email || '';
    const phoneHtml = telephone
        ? `<a class="medecin-conseil-link" href="tel:${encodeURIComponent(telephone)}">${escapeMedecinConseilHtml(telephone)}</a>`
        : '';
    const emailHtml = email
        ? `<a class="medecin-conseil-link" href="mailto:${encodeURIComponent(email)}">${escapeMedecinConseilHtml(email)}</a>`
        : '';
    const contactsHtml = (phoneHtml || emailHtml)
        ? `<div class="medecin-conseil-contacts">${phoneHtml}${emailHtml}</div>`
        : '<p class="muted">Aucun contact n’est encore associé à cette destination.</p>';

    return `
        <article class="medecin-conseil-card">
            <div class="medecin-conseil-card__title">${escapeMedecinConseilHtml(nom)}</div>
            <div class="medecin-conseil-card__destination">${escapeMedecinConseilHtml(destination)}</div>
            ${contactsHtml}
        </article>
    `;
}

function renderMedecinConseilSection(items, { fromCache = false, emptyMessage = null } = {}) {
    const hint = fromCache
        ? 'Disponible hors ligne — coordonnées de la destination choisie à la souscription.'
        : 'Coordonnées du médecin-conseil associé à votre destination de voyage.';
    if (!items || items.length === 0) {
        return `
            <section class="medecin-conseil-section">
                <h3>Médecin-conseil</h3>
                <p class="muted">${escapeMedecinConseilHtml(emptyMessage || 'Aucun médecin-conseil n’est encore associé à une destination de vos souscriptions.')}</p>
            </section>
        `;
    }
    return `
        <section class="medecin-conseil-section">
            <h3>Médecin-conseil</h3>
            <p class="muted">${escapeMedecinConseilHtml(hint)}</p>
            ${items.map(renderMedecinConseilCard).join('')}
        </section>
    `;
}
